client/src/extension.ts

```diff
@@ -51,8 +61,210 @@ export function activate(context: ExtensionContext) {
+	vscode.workspace.onDidChangeTextDocument(async (event) => {
+		if (event.document.languageId === 'parasail') {
+			try{
+				// send the content to the lsp for error checking
+				const errors: { range: { start: { line: number, character: number }, end: { line: number, character: number } }, message: string, severity: vscode.DiagnosticSeverity }[] = await client.sendRequest('parasail.checkErrors', {
+					textDocument: {
+						uri: event.document.uri.toString()},
+				});
+				
+				if (errors) {
+					const diagnostics: vscode.Diagnostic[] = errors.map((error: any) => {
+						const range = new vscode.Range(
+							new vscode.Position(error.range.start.line, error.range.start.character),
+							new vscode.Position(error.range.end.line, error.range.end.character)
+						);
+						const diagnostic = new vscode.Diagnostic(range, error.message, error.severity);
+						return diagnostic;
+					});
+
+					// set the diagnostics 
+					diagnosticCollection.set(event.document.uri, diagnostics);
+				}
+				else{
+					// clear the diagnostics if no errors
+					diagnosticCollection.delete(event.document.uri);
+				}
+			} catch (error) {
+				console.error('Error checking code:', error);
+			}
+		}
+	});
+	//
+	//
+
+	// PLUGIN SHOWDEF FEATURE : uc-plugin-showdef
+	// lsp handles the show definition on hover
+
+	////// not sure how to finish this part, might need to get more information from lsp side?
+	// register hover provider for parasail
+	// const hoverProvider = vscode.languages.registerHoverProvider('parasail', {
+	// 	provideHover: async (document: vscode.TextDocument,
+	// 		position: vscode.Position) => {
+	// 		try {
+	// 			// send the position of hovered text to the lsp
+	// 			const result = await client.sendRequest('parasail.getDefinition', {
+	// 				textDocument: {
+	// 					uri: document.uri.toString()
+	// 				},
+	// 				position: {
+	// 					line: position.line,
+	// 					character: position.character
+	// 				}
+	// 			});
+
+	// 			// if hover information is returned
+	// 			if (result) {
+	// 				const { range, uri, name } = result;
+	// 				const definitionLocation = new vscode.Location(
+	// 					vscode.Uri.parse(uri), new vscode.Range(
+	// 						range.start.line,
+	// 						range.start.character,
+	// 						range.end.line, 
+	// 						range.end.character
+	// 					)
+	// 				);
+					
+	// 				// return the hover information
+	// 				return new vscode.Hover('Definition: ' + name, definitionLocation);
+	// 			} else {
+	// 				// if no hover information is returned
+	// 				return new vscode.Hover('No definition found.');
+	// 			}
+	// 		} catch (error) {
+	// 			console.error('Error fetching definition:', error);
+	// 			return new vscode.Hover('Error fetching definition.');
+	// 		}
+	// 	},
+	// });
+	//
+	//
+
+
+	// add the commands, providers and diagnostic collection to the context
+	context.subscriptions.push(addLibraryCommand, removeLibraryCommand, completionProvider, hoverProvider, diagnosticCollection);

```

Comment:
Don't forget to remove any removed commented out or removed variables from the subscriptions (e.g. `diagnosticCollection`).